index.js

```diff
@@ -143,44 +139,75 @@ ParallelHook.prototype[$map] = function(f){
   return ParallelHook(map(f)(this.hook));
 }
 
-/* c8 ignore next */
-const crash = e => Future(() => { throw e });
-
-ParallelHook.prototype[$ap] = function(mf){
-  return ParallelHook(Hook(c => {
-    let consume = noop;
-    const rf = mf.hook.run(f => consume !== noop ? consume (f) : (
-      Future((rej, res) => {
-        consume = x => map(y => (res(y), y))(c(f(x)));
-        return noop;
-      })
+const CustomFuture = function (interpret) {
+  this._interpret = interpret;
+}
```

Comment:
The fact that this lib consistently needs to rely on Fluture internals makes me consider merging it with the Fluture project.